programs/issue-3-speed-converter/run.py

```diff
@@ -0,0 +1,78 @@
+def m_s_to_km_h(value):
+    converted = value * 3.6
+    return "{} m/s to {} km/h".format(value, converted)
+
+
+def km_h_to_m_s(value):
+    converted = value / 3.6
+    return "{} km/h to {} m/s".format(value, converted)
+
+
+def mile_h_to_km_h(value):
+    converted = value * 1.609
+    return "{} mile/h to {} km/h".format(value, converted)
+
+
+def km_h_to_mile_h(value):
+    converted = value / 1.609
+    return "{} km/h to {} mile/h".format(value, converted)
+
+
+def km_h_to_percent_light_speed(value):
+    converted = value / 1.079e+9
+    return "{} km/h to {}% speed of light".format(value, converted)
+
+
+def end_check():
+    while True:
+        user_options = input("Czy zakończyć program ([T]ak/[N]ie)? ")
+        upper_user_options = user_options.upper()
+        if upper_user_options == "TAK" or upper_user_options == "T":
+            return True
+        elif upper_user_options == "NIE" or upper_user_options == "N":
+            return False
+
+
+def menu():
+    while True:
+        print("Jaką konwersje chcesz wykonać?")
+        print("1) m/s na km/h")
+        print("2) km/h na m/s")
+        print("3) mile/h na km/h")
+        print("4) km/h na mile/h")
+        print("5) km/h na procent prędkości światła w próźni")
+        while True:
+            try:
+                user_options = int(input("Wybierz: "))
+            except ValueError:
+                print("Musisz podać numer opcji!")
+                continue
+            if not 1 <= user_options <= 5:
+                print("Nie istnieje taka opcja.")
+                continue
+            else:
+                break
+        while True:
+            try:
+                value = float(input("Podaj wartość: "))
+            except ValueError:
+                print("Musisz podać liczbę!")
+                continue
+            break
+        if user_options == 1:
+            answer = m_s_to_km_h(value)
+        elif user_options == 2:
+            answer = km_h_to_m_s(value)
+        elif user_options == 3:
+            answer = mile_h_to_km_h(value)
+        elif user_options == 4:
+            answer = km_h_to_mile_h(value)
+        elif user_options == 5:
+            answer = km_h_to_percent_light_speed(value)
+        print(answer)
```

Comment:
Twój kod wydaje się w porządku (nie uruchamiałem go jeszcze), natomiast w tej implementacji jest wada, która w miarę jak trzeba było by dodawać więcej opcji konwersji... bedzie wymagała dodawania coraz to większej liczby ifów.

Dobry kod powinien był łatwo rozszerzalny, tzn. dodanie wsparcia dla nowej funkcji konwertującej, nie powinno wymagać za wielu zmian w innych miejsach. A najlepiej, żeby nie wymagało takich zmian wcale.

Pomyśl jak działają wtyczki do różnych programów. Twórca głównego oprogramowania nie musi dodawać nowych rzeczy do swojego programu, by ten program był kompatybilny z nową wtyczką. To jest kierunek w jakim należy iść.

Podpowiedź:

```
convert_functions = {
 1: m_s_to_km_h,
 2: km_h_to_m_s,
 3: mile_h_to_km_h,
 4: km_h_to_mile_h,
 5: km_h_to_percent_light_speed,
}
```
To jest jeden ze sposobów, który może Cię naprowadzić na dobrą drogę. Ten sposób nadal nie będzie idealny, bo dodanie nowej funkcji dalej będzie wymagać aktualizacji słownika `convert_functions`, ale jest to napewno krok w dobrym kierunku.